tests/TestCase/ORM/Association/HasManyTest.php

```diff
@@ -680,6 +680,51 @@ public function testValueBinderUpdateOnSubQueryStrategy()
         $this->assertCount(2, $authorsAndArticles->get('articles'));
     }
 
+    /**
+     * Tests using subquery strategy when parent query
+     * that contains limit without order.
+     */
+    public function testSubqueryWithLimit()
+    {
+        $Authors = $this->getTableLocator()->get('Authors');
+        $Authors->hasMany('Articles', [
+            'strategy' => Association::STRATEGY_SUBQUERY,
+        ]);
+
+        $query = $Authors->find();
+        $result = $query
+            ->contain('Articles')
+            ->first();
+
+        if (in_array($result->name, ['mariano', 'lary'])) {
+            $this->assertNotEmpty($result->articles);
+        } else {
+            $this->assertEmpty($result->articles);
```

Comment:
This depends on the db since there is no order. Hard to force a result in the middle of these unit tests.